extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/common/MongoDatabaseResolver.java

```diff
@@ -0,0 +1,8 @@
+package io.quarkus.mongodb.panache.common;
+
+/*
+ * This interface can be used to resolve the mongo database name in runtime, it'll help to implement multi-tenancy using tenant per database approach
+ */
+public interface MongoDatabaseResolver {
+    String resolve();
```

Comment:
What still eludes me is how this is actually going to help resolve tenancy.  Without some context, there's not much to be done here.  Is the idea that this resolver would read a different config file?  How are multiple tenants supported by this?

Reply:
For example I would implement it like:
In my API I receive a request header 'x-customer-id' identifying the customer, for each request I'll get the database and put into a ThreadLocal, so the MongoDatabaseResolver could be like:
`@ApplicationScoped
public class CustomMongoCollectionPrefixResolver implements MongoDatabaseResolver {

  @Override
  public String resolve() {
    return DatabaseContext.databaseName.get();
  }
  
}`
In this scenario the database would be selected dynamically

Reply:
fair enough

Reply:
@evanchooly It's missing a new blank line at the end. Should I commit it again?